crates/voicevox_core_c_api/src/lib.rs

```diff
@@ -692,6 +714,208 @@ pub extern "C" fn voicevox_error_result_to_message(
+///
+/// # Safety
+/// @param user_dict は有効な :VoicevoxUserDict のポインタであること
+/// @param word_uuid は呼び出し側で解放する必要がある
+///
+#[no_mangle]
+pub unsafe extern "C" fn voicevox_dict_add_word(
+    user_dict: &VoicevoxUserDict,
+    word: &VoicevoxUserDictWord,
+    out_word_uuid: NonNull<*mut c_char>,
+) -> VoicevoxResultCode {
+    into_result_code_with_error((|| {
+        let word = word.try_into_word()?;
+        let uuid = {
+            let mut dict = user_dict.dict.lock().expect("lock failed");
+            dict.add_word(word)?
+        };
+        let uuid = CString::new(uuid).expect("\\0を含まない文字列であることが保証されている");
+        out_word_uuid
+            .as_ptr()
+            .write_unaligned(C_STRING_DROP_CHECKER.whitelist(uuid).into_raw());
+
+        Ok(())
+    })())
+}
+
+/// ユーザー辞書の単語を更新する
+/// @param [in] user_dict VoicevoxUserDictのポインタ
+/// @param [in] word_uuid 更新する単語のUUID
+/// @param [in] word 新しい単語のデータ
+/// @return 結果コード #VoicevoxResultCode
+///
+/// # Safety
+/// @param user_dict は有効な :VoicevoxUserDict のポインタであること
+#[no_mangle]
+pub unsafe extern "C" fn voicevox_dict_update_word(
+    user_dict: &VoicevoxUserDict,
+    word_uuid: *const u8,
+    word: &VoicevoxUserDictWord,
+) -> VoicevoxResultCode {
+    into_result_code_with_error((|| {
+        let word_uuid = ensure_utf8(unsafe { CStr::from_ptr(word_uuid as *const c_char) })?;
+        let word = word.try_into_word()?;
+        {
+            let mut dict = user_dict.dict.lock().expect("lock failed");
+            dict.update_word(word_uuid, word)?;
+        };
+
+        Ok(())
+    })())
+}
+
+/// ユーザー辞書から単語を削除する
+/// @param [in] user_dict VoicevoxUserDictのポインタ
+/// @param [in] word_uuid 削除する単語のUUID
+/// @return 結果コード #VoicevoxResultCode
+#[no_mangle]
+pub extern "C" fn voicevox_dict_remove_word(
+    user_dict: &VoicevoxUserDict,
+    word_uuid: *const u8,
+) -> VoicevoxResultCode {
+    into_result_code_with_error((|| {
+        let word_uuid = ensure_utf8(unsafe { CStr::from_ptr(word_uuid as *const c_char) })?;
+        {
+            let mut dict = user_dict.dict.lock().expect("lock failed");
+            dict.remove_word(word_uuid)?;
+        };
+
+        Ok(())
+    })())
+}
+
+/// ユーザー辞書の単語をJSON形式で出力する
+/// @param [in] user_dict VoicevoxUserDictのポインタ
+/// @param [out] out_json JSON形式の文字列
+/// @return 結果コード #VoicevoxResultCode
+///
+/// # Safety
+/// @param user_dict は有効な :VoicevoxUserDict のポインタであること
+#[no_mangle]
+pub unsafe extern "C" fn voicevox_dict_get_words_json(
+    user_dict: &VoicevoxUserDict,
+    out_json: NonNull<*mut c_char>,
+) -> VoicevoxResultCode {
+    let dict = user_dict.dict.lock().expect("lock failed");
+    let json = serde_json::to_string(&dict.words()).expect("should be always valid");
+    let json = CString::new(json).expect("\\0を含まない文字列であることが保証されている");
+    out_json
+        .as_ptr()
+        .write_unaligned(C_STRING_DROP_CHECKER.whitelist(json).into_raw());
+    VoicevoxResultCode::VOICEVOX_RESULT_OK
+}
+
+/// 2つのユーザー辞書をマージする
```

Comment:
この説明だと辞書Aと辞書Bをマージして辞書Cを作る感じになりそうかも。
こうとかどうでしょ。
```suggestion
/// 別のユーザー辞書をインポートする
```